src/MessageSource.php

```diff
@@ -31,13 +32,16 @@ public function __construct(
      */
     public function getMessage(string $id, string $category, string $locale, array $parameters = []): ?string
     {
-        $this->bindDomain($category);
-        $this->setLocale($locale);
-        $n = current($parameters);
-        if (is_int($n) === false) {
-            return dgettext($category, $id);
+        $result = $this->getMessageInternal($id, $category, $locale, $parameters);
+        if (!$this->strictMode) {
+            return $result;
         }
-        return dngettext($category, $id, $id, $n);
+
+        if ($result === $id) {
```

Comment:
Result may by equals ID, in this case should be `$result` is returned.

Reply:
You right.